# WRITE OPERATION
# New File Operation

# Clean up of existing file
if "testfile.txt.a" in listfiles():
  removefile("testfile.txt.a")
if "testfile.txt.b" in listfiles():
  removefile("testfile.txt.b")

# Open File Function Call
myfile=ABopenfile("testfile.txt",True)  #Create an AB file

try :
# Write valid data to the file
  myfile.writeat("SXESE8",0)

  try :
        assert('S' == myfile.readat(1,0))
        assert('E' == myfile.readat(None,0)[-1])
  except:
         log(" file is corrupted : file content on Read:" + myfile.readat(None,0))



#Close the file
  myfile.close()

# READ OPERATION
# Reopen file again to read
  myfile=ABopenfile("testfile.txt",True)

# Read the file to check the contents
  try:
      assert('S' == myfile.readat(1,0))
      assert('E' == myfile.readat(None,0)[-1])
     #Close the file
  except:
        #Close the file
        log(" file is corrupted : file content on Read:" + myfile.readat(None,0))
        myfile.close()
        log("Valid Data write to a file is unsuccessfull!")

  try :
# Write valid data to the file
        myfile.writeat("SXdfffdgdfgfdgdfgdfg",4)
        assert('S' == myfile.readat(1,0))
        assert('E' == myfile.readat(None,0)[-1])
        myfile.writeat("SXdfffdgdfgfdgdfgdfgESE8fgddgdggE",4)
        assert('S' == myfile.readat(1,0))
        assert('E' == myfile.readat(None,0)[-1])
  except RepyArgumentError :
            log (" incorrect exception thrown")
  except FileClosedError:
            log( " file closed error even when file is open for reading" )
  except SeekPastEndOfFileError :
        log(" locked forever : reference model does not handle lock release propelry## lock not released on exception #### press CTRl+C to manually terminate" )
  except:
         log("file is corrupted : file content on Read:" + myfile.readat(None,0))

  myfile.close()


except:
 #Close the file
 log("**************checking in exception block*********")
 myfile.close()
 log("Valid Data write to a file is unsuccessfull!")
